strawberryfields/api/connection.py

```diff
@@ -0,0 +1,284 @@
+        response = requests.post(
+            self._url(path),
+            headers=self._headers,
+            data=json.dumps({"circuit": circuit}),
+        )
+        if response.status_code == 201:
+            if self._verbose:
+                log.info("The job was successfully submitted.")
+            return Job(
+                id_=response.json()["id"],
+                status=JobStatus(response.json()["status"]),
+                connection=self,
+            )
+        raise RequestFailedError(
+            "Failed to create job: {}".format(self._format_error_message(response))
+        )
+
+    def get_all_jobs(self, after: datetime = datetime(1970, 1, 1)) -> List[Job]:
+        """Gets a list of jobs created by the user, optionally filtered by datetime.
+
+        Args:
+            after (datetime.datetime): if provided, only jobs more recently created
+                then ``after`` are returned
+
+        Returns:
+            List[strawberryfields.engine.Job]: the jobs
+        """
+        raise NotImplementedError("This feature is not yet implemented")
+
+    def get_job(self, job_id: str) -> Job:
+        """Gets a job.
+
+        Args:
+            job_id (str): the job ID
+
+        Returns:
+            strawberryfields.engine.Job: the job
+        """
+        path = "/jobs/{}".format(job_id)
+        response = requests.get(self._url(path), headers=self._headers)
+        if response.status_code == 200:
+            return Job(
+                id_=response.json()["id"],
+                status=JobStatus(response.json()["status"]),
+                connection=self,
+            )
+        raise RequestFailedError(
+            "Failed to get job: {}".format(self._format_error_message(response))
+        )
+
+    def get_job_status(self, job_id: str) -> JobStatus:
+        """Returns the status of a job.
+
+        Args:
+            job_id (str): the job ID
+
+        Returns:
+            strawberryfields.engine.JobStatus: the job status
+        """
+        return JobStatus(self.get_job(job_id).status)
+
+    def get_job_result(self, job_id: str) -> Result:
+        """Returns the result of a job.
+
+        Args:
+            job_id (str): the job ID
+
+        Returns:
+            strawberryfields.engine.Result: the job result
+        """
+        path = "/jobs/{}/result".format(job_id)
+        response = requests.get(
+            self._url(path), headers={"Accept": "application/x-numpy", **self._headers},
+        )
+        if response.status_code == 200:
+            # Read the numpy binary data in the payload into memory
+            with io.BytesIO() as buf:
+                buf.write(response.content)
+                buf.seek(0)
+                samples = np.load(buf)
+            return Result(samples, is_stateful=False)
+        raise RequestFailedError(
+            "Failed to get job result: {}".format(self._format_error_message(response))
+        )
+
+    def cancel_job(self, job_id: str):
```

Comment:
Better to use `id_` as used elsewhere?

Reply:
I think using `id_` within the `Job` class itself makes sense, since it is quite obvious what the variable is identifying; but I think `job_id` is more explicit for methods on other classes and makes the interface a bit clearer.